### PR TITLE
Add: standalone Qwen3 HBG decode benchmark

### PR DESCRIPTION
**Pull request by Crane-Liu**
## Summary

Adds a manual Qwen3-14B decode benchmark for `host_build_graph`, with single-slot and dual-slot entrypoints using caller-provided weights, prefill KV snapshots, and golden tokens.

The artifact adapter outlines the generated 40-layer decoder into one bounded per-layer Definition: 277 tasks, 28 tensor boundaries, and 40 invocations per frame. It preserves in-core binaries and both runners verify artifact checksums before preparing the runtime. The adapter accepts the 25-argument ABI and the 26-argument ABI with `sampled_ids_host`.

STRACE extraction reports RTS completion interval, `runner_run`, Effective (`device_wall`), `graph_build`, and bind timing. It exports `simpler_strace_timeline.json` for Perfetto. Qualification defaults to zero warmups, one measured request, 127 decode dispatches, and steady skip 5; the Python entrypoints also support 1 warmup + 5 measured runs.

## Validation

- Based on `main` at `6940c8ccbc4d8ec7111e542a0ce4a4985a99a24a`.
- 14 CPU unit tests pass, covering slot isolation, ABI checks, artifact integrity, and timing boundaries.
- Native artifact builds with all 39 in-core binaries byte-identical to the supplied source artifact.
- A2/A3 single-slot and dual-slot 8-dispatch hardware smoke passes. Token rows match golden and each other; dual slots alternate with independent generations and prepared dispatches. Both Perfetto exports have eight aligned device spans.
- Full repository pre-commit checks pass for all changed files.

Full 127-dispatch qualification on the latest main remains incomplete: a previous main candidate with the supplied legacy 25-argument artifact failed strict golden comparison. The cause is unresolved. The 26-argument path has metadata/unit coverage and still requires hardware qualification.

### CODE REVIEW

File: examples/a2a3/host_build_graph/qwen3_14b_serving_effective/benchmark.py [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+    compiled: Any,
+    run_config: Any,
+    param_names: list[str],
+    resident: dict[str, Any],
+    slots: list[dict[str, torch.Tensor]],
+    golden: dict[str, torch.Tensor],
+    mode: str,
+    steps: int,
+    sampled_ids_host_abi: bool,
+) -> dict[str, Any]:
```
[the remark this answers is not in the document]
Comment (Crane-Liu) [reply examples/a2a3/host_build_graph/qwen3_14b_serving_effective/benchmark.py]: Fixed — shared slot and geometry helpers now live in hbg_common.py, and single/dual entrypoints use the shared contract.

File: examples/a2a3/host_build_graph/qwen3_14b_serving_effective/build_hbg_artifact.py [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+        source,
+    )
+    if loop_match is None:
+        raise RuntimeError("cannot locate generated 40-layer loop")
+    layer_index = loop_match.group("layer")
+    loop_open = source.find("{", loop_match.start())
+    loop_close = _brace_end(source, loop_open)
+    scope_start = source.find("SIMPLER_SCOPE()", loop_open, loop_close)
+    scope_open = source.find("{", scope_start, loop_close)
+    scope_close = _brace_end(source, scope_open)
```
[the remark this answers is not in the document]
Comment (Crane-Liu) [reply examples/a2a3/host_build_graph/qwen3_14b_serving_effective/build_hbg_artifact.py:233]: Fixed — the generated SIMPLER_SCOPE marker and its opening brace are validated before block parsing.

File: examples/a2a3/host_build_graph/qwen3_14b_serving_effective/fixture.py [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+            raise ValueError(f"invalid SHA256SUMS path: {relative}")
+        if relative_path in expected:
+            raise ValueError(f"duplicate SHA256SUMS path: {relative_path}")
+        expected[relative_path] = digest
+    actual = {path.relative_to(root) for path in root.rglob("*") if path.is_file() and path.name != "SHA256SUMS"}
+    if set(expected) != actual:
+        raise ValueError(
+            f"SHA256SUMS file set mismatch: missing={sorted(actual - set(expected))} "
+            f"extra={sorted(set(expected) - actual)}"
+        )
```
[the remark this answers is not in the document]
Comment (Crane-Liu) [reply examples/a2a3/host_build_graph/qwen3_14b_serving_effective/fixture.py:117]: Fixed — SHA256SUMS mismatch diagnostics now label expected entries as missing and on-disk extras correctly.

File: examples/a2a3/host_build_graph/qwen3_14b_serving_effective/trace_effective.py [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+
+
+def invocation_rows(spans: list[Span]) -> list[dict[str, float | int | bool | str]]:
+    roots = [span for span in spans if span.name in {"chip.run", "simpler_run"}]
+    if not roots:
+        raise ValueError("no chip.run or simpler_run root spans found")
+
+    spans_by_inv: dict[int, list[Span]] = {}
+    for span in spans:
+        spans_by_inv.setdefault(span.inv, []).append(span)
```
[the remark this answers is not in the document]
Comment (Crane-Liu) [reply examples/a2a3/host_build_graph/qwen3_14b_serving_effective/trace_effective.py]: Fixed — trace groups remain process-scoped and child runtime spans are associated with the matching root run.

File: examples/a2a3/host_build_graph/qwen3_14b_serving_effective/validate_dual_result.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+def main() -> int:
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("--result", type=Path, required=True)
+    parser.add_argument("--fixture", type=Path, required=True)
+    parser.add_argument("--golden-manifest", type=Path, required=True)
+    parser.add_argument("--steps", type=int, required=True)
+    args = parser.parse_args()
+
+    benchmark = json.loads((args.result / "benchmark.json").read_text())
+    trace = json.loads((args.result / "trace_summary.json").read_text())
```
[the remark this answers is not in the document]
Comment (Crane-Liu) [reply examples/a2a3/host_build_graph/qwen3_14b_serving_effective/validate_dual_result.py:47]: Fixed — dual qualification recomputes aggregation and official metrics from run.log before writing validation output.

File: examples/a2a3/host_build_graph/qwen3_14b_serving_effective/weights.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+from pathlib import Path
+
+import torch
+from safetensors import safe_open
+
+NUM_LAYERS = 40
+HEAD_DIM = 128
+HIDDEN = 5120
+PADDED_VOCAB = 152064
+MAX_SEQ_LEN = 4096
```
[the remark this answers is not in the document]
Comment (Crane-Liu) [reply examples/a2a3/host_build_graph/qwen3_14b_serving_effective/weights.py:25]: Fixed — checkpoint geometry is validated against the Qwen decode ABI before weight conversion.

**Review by ChaoZheng109 — APPROVED** [review]
